### PR TITLE
Publishing for haydentherapper!

### PR DESCRIPTION
**Pull request by Hayden-IO**
Signed-off-by: Hayden Blauzvern <hblauzvern@google.com>

#### Ticket Link

Fixes

#### Release Note

```release-note

```

### CODE REVIEW

**Comment by Hayden-IO** [comment]
```
VERIFYING TUF CLIENT UPDATE

Client successfully initialized, updating and downloading targets...
Client updated to...
	root.json version 3, expires 2022/11/10
	timestamp.json version 27, expires 2022/05/25
	snapshot.json version 27, expires 2022/05/25
	targets.json version 3, expires 2022/11/10

Retrieved target fulcio.crt.pem...
-----BEGIN CERTIFICATE-----
MIIB+DCCAX6gAwIBAgITNVkDZoCiofPDsy7dfm6geLbuhzAKBggqhkjOPQQDAzAq
MRUwEwYDVQQKEwxzaWdzdG9yZS5kZXYxETAPBgNVBAMTCHNpZ3N0b3JlMB4XDTIx
MDMwNzAzMjAyOVoXDTMxMDIyMzAzMjAyOVowKjEVMBMGA1UEChMMc2lnc3RvcmUu
ZGV2MREwDwYDVQQDEwhzaWdzdG9yZTB2MBAGByqGSM49AgEGBSuBBAAiA2IABLSy
A7Ii5k+pNO8ZEWY0ylemWDowOkNa3kL+GZE5Z5GWehL9/A9bRNA3RbrsZ5i0Jcas
taRL7Sp5fp/jD5dxqc/UdTVnlvS16an+2Yfswe/QuLolRUCrcOE2+2iA5+tzd6Nm
MGQwDgYDVR0PAQH/BAQDAgEGMBIGA1UdEwEB/wQIMAYBAf8CAQEwHQYDVR0OBBYE
FMjFHQBBmiQpMlEk6w2uSu1KBtPsMB8GA1UdIwQYMBaAFMjFHQBBmiQpMlEk6w2u
Su1KBtPsMAoGCCqGSM49BAMDA2gAMGUCMH8liWJfMui6vXXBhjDgY4MwslmN/TJx
Ve/83WrFomwmNf056y1X48F9c4m3a3ozXAIxAKjRay5/aj/jsKKGIkmQatjI8uup
Hr/+CxFvaJWmpYqNkLDGRU+9orzh5hI2RrcuaQ==
-----END CERTIFICATE-----

Retrieved target fulcio_v1.crt.pem...
-----BEGIN CERTIFICATE-----
MIIB9zCCAXygAwIBAgIUALZNAPFdxHPwjeDloDwyYChAO/4wCgYIKoZIzj0EAwMw
KjEVMBMGA1UEChMMc2lnc3RvcmUuZGV2MREwDwYDVQQDEwhzaWdzdG9yZTAeFw0y
MTEwMDcxMzU2NTlaFw0zMTEwMDUxMzU2NThaMCoxFTATBgNVBAoTDHNpZ3N0b3Jl
LmRldjERMA8GA1UEAxMIc2lnc3RvcmUwdjAQBgcqhkjOPQIBBgUrgQQAIgNiAAT7
XeFT4rb3PQGwS4IajtLk3/OlnpgangaBclYpsYBr5i+4ynB07ceb3LP0OIOZdxex
X69c5iVuyJRQ+Hz05yi+UF3uBWAlHpiS5sh0+H2GHE7SXrk1EC5m1Tr19L9gg92j
YzBhMA4GA1UdDwEB/wQEAwIBBjAPBgNVHRMBAf8EBTADAQH/MB0GA1UdDgQWBBRY
wB5fkUWlZql6zJChkyLQKsXF+jAfBgNVHSMEGDAWgBRYwB5fkUWlZql6zJChkyLQ
KsXF+jAKBggqhkjOPQQDAwNpADBmAjEAj1nHeXZp+13NWBNa+EDsDP8G1WWg1tCM
WP/WHPqpaVo0jhsweNFZgSs0eE7wYI4qAjEA2WB9ot98sIkoF3vZYdd3/VtWB5b9
TNMea7Ix/stJ5TfcLLeABLE4BNJOsQ4vnBHJ
-----END CERTIFICATE-----

Retrieved target rekor.0.pub...
-----BEGIN PUBLIC KEY-----
MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAE2G2Y+2tabdTV5BcGiBIx0a9fAFwr
kBbmLSGtks4L3qX6yYY0zufBnhC8Ur/iy55GhWP/9A/bY2LhC30M9+RYtw==
-----END PUBLIC KEY-----

Retrieved target rekor.pub...
-----BEGIN PUBLIC KEY-----
MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAE2G2Y+2tabdTV5BcGiBIx0a9fAFwr
kBbmLSGtks4L3qX6yYY0zufBnhC8Ur/iy55GhWP/9A/bY2LhC30M9+RYtw==
-----END PUBLIC KEY-----

Retrieved target artifact.pub...
-----BEGIN PUBLIC KEY-----
MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEhyQCx0E9wQWSFI9ULGwy3BuRklnt
IqozONbbdbqz11hlRJy9c7SG+hdcFl9jE9uE/dwtuwU2MqU9T/cN0YkWww==
-----END PUBLIC KEY-----

Retrieved target ctfe.pub...
-----BEGIN PUBLIC KEY-----
MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEbfwR+RJudXscgRBRpKX1XFDy3Pyu
dDxz/SfnRi1fT8ekpfBd2O1uoz7jr3Z8nKzxA69EUQ+eFCFI3zeubPWU7w==
-----END PUBLIC KEY-----
```

**Comment by Hayden-IO** [comment]
Please check:
* All metadata files look correct
* Metadata updates are limited to version, expiration, and signature
* Nothing has been unexpectedly moved around

File: repository/repository/targets.json [hunk 1]
```diff
@@ -104,6 +96,13 @@
 				},
 				"length": 740
 			},
+			"rekor.0.pub": {
```
Comment (Hayden-IO) [inline repository/repository/targets.json:99]: This doesn't look correct, the Rekor delegation should sign off on the rekor shard key - Are we ok with this for now?

Comment (Hayden-IO) [reply repository/repository/targets.json:99]: This was also present in the v3 root signing.

It's going to be a lot of work to re-sign the targets at this point. I think we should just resolve this with the next root signing.

**Comment by Hayden-IO** [comment]
After pushing the metadata to the remote:

```
$ cosign initialize --mirror v3-test-tuf-root-staging --root ~/cosign/pkg/cosign/tuf/repository/root.json
Root status:
 {
	"local": "/usr/local/google/home/hblauzvern/.sigstore/root",
	"remote": "v3-test-tuf-root-staging",
	"metadata": {
		"root.json": {
			"version": 3,
			"len": 4810,
			"expiration": "10 Nov 22 21:58 UTC",
			"error": ""
		},
		"snapshot.json": {
			"version": 27,
			"len": 1657,
			"expiration": "25 May 22 00:23 UTC",
			"error": ""
		},
		"targets.json": {
			"version": 3,
			"len": 3972,
			"expiration": "10 Nov 22 21:58 UTC",
			"error": ""
		},
		"timestamp.json": {
			"version": 27,
			"len": 719,
			"expiration": "25 May 22 00:23 UTC",
			"error": ""
		}
	},
	"targets": [
		"rekor.pub",
		"artifact.pub",
		"ctfe.pub",
		"fulcio.crt.pem",
		"fulcio_v1.crt.pem",
		"rekor.0.pub"
	]
}
```

**Comment by Hayden-IO** [comment]
Testing with the embedded metadata, with the local TUF DB deleted:

```
$ cat ~/.sigstore/root/remote.json
{"mirror":"v3-test-tuf-root-staging"}

COSIGN_EXPERIMENTAL=1 cosign verify ...

Verification for ....
```

**Review by dlorenc — APPROVED** [review]

**Comment by trishankatdatadog** [comment]
Just seeing this: this is so cool!

**Pull request merged**